packages/gatsby-plugin-utils/src/test-plugin-options-schema.ts

```diff
@@ -44,11 +46,23 @@ export async function testPluginOptionsSchema(
   })
 
   try {
-    await validateOptionsSchema(pluginSchema, pluginOptions)
+    const { warning } = await validateOptionsSchema(pluginSchema, pluginOptions)
+
+    console.log(`warning`, warning)
+    const warnings = warning?.details?.map(detail => detail.message) ?? []
+
+    if (warnings?.length > 0) {
+      return {
+        isValid: true,
+        errors: [],
+        hasWarnings: true,
```

Comment:
Could we omit `hasWarnings` since we're returning `warnings` here?

If we introduce `hasWarnings` I'd also expect a `hasErrors`, which we don't have. We can keep the pattern for testing warnings the same as errors this way with `expect(warnings).toEqual(expectedWarnings)` (like you've done above).

Reply:
`hasWarning` is the corollary to `isValid`. I was trying not to modify the existing naming system and potentially breaking more things. I could eliminate hasWarnings in favor of `errors.length === 0` but this seemed simpler.

Reply:
How would you like me to handle this?

Reply:
I see your logic, it also makes sense. Let's keep this as is.